rfp-proposal/on-chain-quadratic-funding.md

```diff
@@ -0,0 +1,68 @@
+
+[Gitcoin](https://gitcoin.co/) is currently using this mechanism to successfully fund and support public goods. However, gitcoin is centralized. The goal of this RFP is to develop a decentralized solution on top of [Substrate](https://github.com/paritytech/substrate), which can potentially be integrated into Kusama and Polkadot. The on-chain treasury could potentially sustainably fund the matching pool in the long-run. However, the Web3 Foundation would also be committed to fund matching pool of the solution for initial test rounds. 
+
+Key-problems to solve as part of the grant:
+1. **Anti-collusion** 
+2. **Sybil resistance**
+3. **User-friendly UI**
+ 
+
+## Deliverables :nut_and_bolt:
+
+The milestones below are just an initial draft. The milestones can be structured completely differently and the implementation can also leverage other tools and infrastructure as long as the overall goal of the RFP is met.  
+
+### Milestone 1 - Anti-collusion
+
+| Number | Deliverable | Specification | 
+| ------------- | ------------- | ------------- |
+| 0a. | License | Apache 2.0 / MIT / Unlicense |
+| 0b. | Documentation | Inline documentation of the code and a basic tutorial that explains how a developer can use the project |
+| 0c. | Testing Guide | The code will have unit-test coverage (min. 70%) to ensure functionality and robustness. In the guide we will describe how to run these tests | 
+| 0d. | Article/Tutorial | Article or tutorial that explains the work done as part of the grant. 
+| 1. | Anti-collusion | Implement a mechanism to prevent bribery and collusion, leveraging encrypting votes (ZKPs) potentially via [Minimum Anti-Collusion Infrastructure (MACI)](https://ethresear.ch/t/minimal-anti-collusion-infrastructure/5413) |  
+| 2.  | Voting Example | Integrate a basic voting example that lets you test the mechanism | 
+
+
+### Milestone 2 - Sybil resistance
+
+| Number | Deliverable | Specification | 
+| ------------- | ------------- | ------------- |
+| 0a. | License | Apache 2.0 / MIT / Unlicense |
+| 0b. | Documentation | Inline documentation of the code and a basic tutorial that explains how a developer can use the project|
+| 0c. | Testing Guide | The code will have unit-test coverage (min. 70%) to ensure functionality and robustness. In the guide we will describe how to run these tests | 
+| 0d. | Article/Tutorial | Article or tutorial that explains the work done as part of the grant. 
+| 1. | Sybil resistance | Mechanism to prevent sybil attacks, either implemented as its own substrate pallet or possible integration of the existing on-chain identity on Kusama/Polkadot provided by specified registrar  |  
+| 2.  | UI | UI for verification of identities, including a polkadot.js extension support | 
+
+### Milestone 3 - CLR
```

Comment:
Shouldn't this milestone come first? The other parts (anti collusion + Sybil protection), while necessary for a real deployment, are not needed to demonstrate the functionality, which IMO should be done as a first step.

Reply:
The problem is you can not really test it before you have a solution for the two problems. Maybe if you use a centralized replacement initially. But probably it's best to just leave it up to the implementers to decide about the order and specifically mention this. I think the implementation of MACI on its one would be a useful project/grant.